File: src/Cxx/Images/StaticImage.cxx
#include <vtkSmartPointer.h>

#include <vtkImageReader2Factory.h>
#include <vtkImageReader2.h>
#include <vtkImageViewer2.h>
#include <vtkRenderWindow.h>
#include <vtkRenderWindowInteractor.h>
#include <vtkRenderer.h>

int main(int argc, char* argv[])
{
  // Verify input arguments
  if ( argc != 2 )
  {
    std::cout << "Usage: " << argv[0]
              << " Filename" << std::endl;
    return EXIT_FAILURE;
  }

  // Read the image
  vtkSmartPointer<vtkImageReader2Factory> readerFactory =
    vtkSmartPointer<vtkImageReader2Factory>::New();
  vtkSmartPointer<vtkImageReader2> reader;
  reader.TakeReference(
    readerFactory->CreateImageReader2(argv[1]));
  reader->SetFileName(argv[1]);
  reader->Update();

  // Visualize
  vtkSmartPointer<vtkImageViewer2> imageViewer =
    vtkSmartPointer<vtkImageViewer2>::New();
  imageViewer->SetInputConnection( reader->GetOutputPort() );
  imageViewer->GetRenderWindow()->SetSize( 500, 500 );
  imageViewer->GetRenderer()->ResetCamera();

  // Set up an interactor that does not respond to mouse events
  vtkSmartPointer<vtkRenderWindowInteractor> renderWindowInteractor =
    vtkSmartPointer<vtkRenderWindowInteractor>::New();
  imageViewer->GetRenderWindow()->SetInteractor( renderWindowInteractor );
  renderWindowInteractor->SetInteractorStyle( 0 );
  imageViewer->Render();

  // Start the event loop
  renderWindowInteractor->Initialize();
  renderWindowInteractor->Start();

  return EXIT_SUCCESS;
}

File: src/Java/Images/StaticImage.java
import vtk.vtkNativeLibrary;
import vtk.vtkRenderWindowInteractor;
import vtk.vtkImageReader2;
import vtk.vtkImageViewer2;
import vtk.vtkImageReader2Factory;

public class StaticImage 
{
  // -----------------------------------------------------------------
  // Load VTK library and print which library was not properly loaded
  static 
  {
    if (!vtkNativeLibrary.LoadAllNativeLibraries()) 
    {
      for (vtkNativeLibrary lib : vtkNativeLibrary.values()) 
      {
        if (!lib.IsLoaded()) 
        {
          System.out.println(lib.GetLibraryName() + " not loaded");
        }
      }
    }
    vtkNativeLibrary.DisableOutputWindow(null);
  }
  // -----------------------------------------------------------------

  public static void main(String args[]) 
  {
    // Read the image (Pass the Image File Name as Argument)
    vtkImageReader2Factory ReaderFactory = new vtkImageReader2Factory();
    vtkImageReader2 Reader = ReaderFactory.CreateImageReader2(args[0]);
    Reader.SetFileName(args[0]);
    Reader.Update();
		
    // Visualize
    vtkImageViewer2 ImageViewer = new vtkImageViewer2();
    ImageViewer.SetInputConnection(Reader.GetOutputPort());
    ImageViewer.GetRenderWindow().SetSize(500, 500);
    ImageViewer.GetRenderer().ResetCamera();

    // Set up an interactor that does not respond to mouse events
    vtkRenderWindowInteractor iren = new vtkRenderWindowInteractor();
    ImageViewer.GetRenderWindow().SetInteractor(iren);
    iren.SetInteractorStyle(null);
    ImageViewer.Render();

    // Start the event loop
    iren.Initialize();
    iren.Start();

    Reader.Delete();
  }
}
